wordpress-mobile/WordPress-FluxC-Android
# Use 'AndroidKeyStore' in the trust manager

Currently `MemorizingTrustManager` stores SSL certs using a key store in the app's file directory with a password that is hardcoded and therefore public. Although it can be tricky to access the file, this is not a secure & robust way of keeping the cert info safe.

I have made a relatively small change here to change the `KeyStore` type to `AndroidKeyStore`. This is the new type introduced in API 18 which allows saving in a system `KeyStore`. Access is limited to the app so cannot be snooped.

Some relevant resources:

- https://developer.android.com/reference/java/security/KeyStore
- https://developer.android.com/training/articles/keystore#UsingAndroidKeyStore
- https://academy.realm.io/posts/secure-storage-in-android-san-francisco-android-meetup-2017-najafzadeh/#android-keystore-system

To test:
- I would like to do some more thorough testing with custom SSL certificates before moving forward.
- Try out migration from the existing version to make sure it works gracefully.

If we go ahead with this approach I will follow up with a change to delete `wpstore_certs_truststore.bks` if it exists. This will ensure no sensitive data is sitting around, even after updating.
